src/WebJobs.Extensions.OpenAI.CosmosDBSearch/CosmosDBSearchProvider.cs

```diff
@@ -0,0 +1,266 @@
+    public async Task<SearchResponse> SearchAsync(SearchRequest request)
+    {
+        if (request.Embeddings.IsEmpty)
+        {
+            throw new ArgumentException("Emeddings must be provided");
+        }
+
+        if (request.ConnectionInfo is null)
+        {
+            throw new ArgumentNullException(nameof(request.ConnectionInfo));
+        }
+
+        string connectionString = this.configuration.GetValue<string>(request.ConnectionInfo.ConnectionName);
+        MongoClient mongoClient = await this.GetMongoClientAsync(connectionString);
+
+        try
+        {
+            IMongoCollection<BsonDocument> collection = mongoClient.GetDatabase(databaseName).GetCollection<BsonDocument>(request.ConnectionInfo.CollectionName ?? defaultSearchIndexName);
+
+            // Search Azure Cosmos DB for MongoDB vCore collection for similar embeddings and project fields.
+            BsonDocument[] pipeline = new BsonDocument[]
+            {
+                new BsonDocument
+                {
+                    {
+                        "$search", new BsonDocument
+                        {
+                            {
+                                "cosmosSearch", new BsonDocument
+                                {
+                                    { "vector", !(request.Embeddings.IsEmpty) ? new BsonArray(request.Embeddings.ToArray()) : new BsonArray()},
+                                    { "path", "embedding" },
+                                    { "k", request.MaxResults }
+                                }
+                            },
+                            { "returnStoredSource", true }
+                        }
+                    }
+                },
+                new BsonDocument
+                {
+                    {
+                        "$project", new BsonDocument
+                        {
+                            { "embedding", 0 },
+                            { "_id", 0 },
+                            { "id", 0 },
+                            { "timestamp", 0 }
+                        }
+                    }
+                }
+            };
+
+            IAsyncCursor<BsonDocument> cursor = await collection.AggregateAsync<BsonDocument>(pipeline);
+            List<BsonDocument> documents = await cursor.ToListAsync();
+
+            var resultFromDocuments = documents.ToList().ConvertAll(bsonDocument => BsonSerializer.Deserialize<OutputDocument>(bsonDocument));
+
+            List<SearchResult> searchResults = new();
+            
+            foreach (OutputDocument doc in resultFromDocuments)
+            {
+                searchResults.Add(new SearchResult(doc.Title, doc.Text));
+            }
+
+            SearchResponse response =  new SearchResponse(searchResults);
+            return response;
+
+        }
+        catch (MongoException ex)
+        {
+            this.logger.LogError($"Exception: SearchAsync(): {ex.Message}");
```

Comment:
We should be logging the full exception and not just the message. I think there's an overload in `LogError` for logging the entire exception object.

Reply:
Same comment for all other places where we're logging exception information.